# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/shared/Lowering-shared.cpp
# Commit: 38d4e11b069b
# Full Hash: 38d4e11b069b609199ed6017ce80e199d2c5773b
# Author: Lars T Hansen <lhansen@mozilla.com>
# Date: 2020-05-20 15:28:23
# Regressor Bug: 1631228
# File Overlap Count: 1
# Description:
#   Bug 1631228 - wasm ion simd, part 3: ion code generation. r=bbouvier
#   
#   Add SIMD opcodes to WasmIonCompile and implement code generation for
#   them on x64.
#   
# ==============================================================================

diff -r 9aacd7b8b252 -r 38d4e11b069b js/src/jit/shared/Lowering-shared.cpp
--- a/js/src/jit/shared/Lowering-shared.cpp	Wed May 20 07:02:49 2020 +0000
+++ b/js/src/jit/shared/Lowering-shared.cpp	Wed May 20 07:03:54 2020 +0000
@@ -300,3 +300,39 @@
     return;
   }
 }
+
+#ifndef ENABLE_WASM_SIMD
+
+void LIRGenerator::visitWasmBitselectSimd128(MWasmBitselectSimd128*) {
+  MOZ_CRASH("SIMD not enabled");
+}
+
+void LIRGenerator::visitWasmBinarySimd128(MWasmBinarySimd128*) {
+  MOZ_CRASH("SIMD not enabled");
+}
+
+void LIRGenerator::visitWasmShiftSimd128(MWasmShiftSimd128*) {
+  MOZ_CRASH("SIMD not enabled");
+}
+
+void LIRGenerator::visitWasmShuffleSimd128(MWasmShuffleSimd128*) {
+  MOZ_CRASH("SIMD not enabled");
+}
+
+void LIRGenerator::visitWasmReplaceLaneSimd128(MWasmReplaceLaneSimd128*) {
+  MOZ_CRASH("SIMD not enabled");
+}
+
+void LIRGenerator::visitWasmScalarToSimd128(MWasmScalarToSimd128*) {
+  MOZ_CRASH("SIMD not enabled");
+}
+
+void LIRGenerator::visitWasmUnarySimd128(MWasmUnarySimd128*) {
+  MOZ_CRASH("SIMD not enabled");
+}
+
+void LIRGenerator::visitWasmReduceSimd128(MWasmReduceSimd128*) {
+  MOZ_CRASH("SIMD not enabled");
+}
+
+#endif  // !ENABLE_WASM_SIMD